### DISCUSSION TITLE
[Feature Proposal] Tailwind CSS Cheat Sheet

### DISCUSSION THREAD

**Discussion by ghost** [tailwindlabs/tailwindcss#2216] [Ideas]
I think it would be very helpful to have a cheat sheet for all of the utility classes and such. Since we have access to all of the utilities in the `.js` files, we could probably create a npm script that could generate the file to display all of the utilities.

Whatcha think? 

p.s. What would also be really helpful is if we could produce a local `.json` cheat sheet file (or some other type) that was specific to that project. That way when one is working on a project, you just switch to that file, do your search for the utility class that you are looking for and then swap back to the file you were working on. This would be great because with most editors, you could simply do a cmd+p -> cheatsheet.json, then cmd+f -> margin... find the utility and bam. No context switching by leaving the editor.

That may be a bit much, but could be helpful.

**Comment by adamwathan** [comment]
This is definitely a great idea and something we've wanted to do from the beginning. Have you seen any other cheat sheets that you think are formatted in a really useful way that we could take inspiration from? Or are you just thinking of a giant table similar to this but with every class:

![image](https://user-images.githubusercontent.com/4323180/33382489-de09463e-d4ee-11e7-97e7-3330849d0139.png)

**Comment by tlgreg** [comment]
I also kinda like, what tachyons does with their overview pages, it's easy to look up a property.
[Table of Styles](http://tachyons.io/docs/table-of-styles/)
[Table of Properties](http://tachyons.io/docs/table-of-properties/)

**Comment by ghost** [comment]
I really like what Tachyons is doing with theirs. When I thought of it, I was thinking more along the lines of this: http://fontawesome.io/cheatsheet/ With that said, I do not think it has to be just like that and I really like the layout that @adamwathan presented. 

it would be cool if it had a search mechanism. I mean, `cmd+f/ctrl+f` works, so that is not a necessity - rather, a luxury.

**Comment by HellPat** [comment]
Mhh, maybe it's possible to index all classes to algolia somehow and add a third section to autocomplete ore something like that.

**Comment by ghost** [comment]
It doesn’t even have to be that in-depth, although, that would be killer. It could be as simple as using JS .filter() 

Deric Cain

**Comment by damian-nz** [comment]
Maybe even just simply grouping the classes by function

**Comment by chrisdmacrae** [comment]
@adamwathan could this be done outside of core?

**Comment by adamwathan** [comment]
Yep totally could I think; someone has actually made one here just not sure if it's autogenerated or put together by hand:

http://nerdcave.com/tailwind-cheat-sheet

**Comment by derekphilipau** [comment]
@ZachJMoore and I wanted to use Tailwind CSS in some projects and thought a cheat sheet would be a great way to learn:  https://github.com/derekphilipau/tailwindcss-cheatsheet-single-page  Zach also made some layout & component examples here:  https://github.com/ZachJMoore/tailwindcss-layout-examples  We hope you find them useful and let us know if you have ideas for improvements!

**Comment by donnysim** [comment]
It would be great if in addition to cheat sheet, tailwind could generate an html file where it displays how everything looks based on tailwind config colors, shadows etc., including ability to add to the generated html for plugins like buttons etc.. Every tailwind config can be different, so having the ability to see the padding's, colors, shadows etc. and what class it is would be great instead of having to hunt the correct value.

**Comment by deadcoder0904** [comment]
Would definitely love this. Currently, I love referring to https://flaviocopes.com/tailwind-cheat-sheet/

**Comment by umeshmk** [comment]
I tried my best to create this cheatsheet for me. 
Please review.
https://umeshmk.github.io/Tailwindcss-cheatsheet/

**Comment by deadcoder0904** [comment]
@umeshmk that indeed looks awesome 👌

**Comment by xavism** [comment]
Any update on this? It should be pretty easy to create an html with the config based on a JSON, if no one is working on this, I could start to do something on my own and publish it to the community

**Comment by findexu** [comment]
For v3 you can find it here: https://tailwindcomponents.com/cheatsheet/

**Comment by eugenefm** [comment]
Any one find a good solution for this? Would love an automated way to put together a cheatsheet with our custom config which uses a distinct color palette, spacing and font scale from the Tailwind default.

**Comment by brittneypostma** [comment]
Also looking for something like this. Something like [Tailwind Config Viewer](https://github.com/rogden/tailwind-config-viewer) is nice, but doesn't necessarily need to build a site. Functions that transformed the classes would be enough.

**Comment by faraz-jb** [comment]
For anyone who wants a cheat sheet generated from *their own* config (custom colors, spacing, fonts) — this is solved: **tailwind-config-viewer** (`npx tailwind-config-viewer`). It serves a local interactive page listing every utility class your project actually has, built straight from your `tailwind.config`. It picks up custom theme values automatically, which is exactly the "project-specific cheat sheet" idea from this thread.

For v4 specifically, the `@theme` CSS variables are the single source of truth — a small script can parse those and generate a JSON/markdown cheat sheet for docs or a searchable tool page.
